pkg/operator/ceph/operator.go

```diff
@@ -227,6 +227,11 @@ func (o *Operator) updateDrivers() error {
 		return errors.Wrap(err, "failed to configure CSI parameters")
 	}
 
+	if !csi.CSIEnabled() {
```

Comment:
@jshen28 Thanks for the patch .looks like this can cause an issue if someone enabled the CSI  driver at first and then later disables it. Currently, if the CSI driver is enabled and then disabled we will delete the CSI pods which are started when the CSI driver was enabled for the first time. but with this change, we return the success but we don't delete the csi driver deployment and the daemonset resources.

I would suggest handling this in ValidateAndConfigureDrivers

Reply:
Right, sorry I ignored this case...